enhancements/kube-apiserver/stale-clusteroperators-detection.md

```diff
@@ -105,4 +105,13 @@ No API extension is needed to implement this proposal.
 One alternative discussed was a new API field that will be added to clusteroperators.Status called "lastUpdateTime".
 Instead of relying on
 "lastTransitionTime" will introduce this field and use it as "heart-beat". This will require **all** operators to update
-this field correctly and multiply the writes significantly.
\ No newline at end of file
+this field correctly and multiply the writes significantly.
+
+Another alternative is to use lease inspection based on explicitly named leases in the .status.relatedResources for each
+clusteroperator.
+This alternative resolves the additional heartbeat problem and doesn't require a logic change since operators already 
+provide .status.relatedResources.
+Every operator would need updating to include their lease and the staleness checker navigates through and inspects leases.
+This is preferred, but takes significantly longer to produce value.
```

Comment:
longer to produce value due to the requirement that each operator deliver a change(to add the lease to the relatedResource), i assume?

we could presumably get some shorter term value from a few operators implementing this, even if we don't immediately get to all of them, but yes it becomes a longer journey.

We could also pursue this path after implementing the flapping status message in the short term.  I.e. implement the flapping status approach, then start chasing adding relatedResources, and once that is done, switch the implementation to use that.  Is that worth considering?

Reply:
It's longer both in adding to every operator and in the logic required for navigation, inspection, and watching.

> We could also pursue this path after implementing the flapping status message in the short term. I.e. implement the flapping status approach, then start chasing adding relatedResources, and once that is done, switch the implementation to use that. Is that worth considering?

I think this is valid, though I think a determination of priority against other pain-points is different in a world where staleness can already be determined. iow, is it still important enough to be worked on after staleness detection is no longer failing.

Reply:
>  is it still important enough to be worked on after staleness detection is no longer failing.

yeah, i agree it's unlikely to get investment in the follow up, but at least that would be an option if we conclude the message flapping is causing confusion/annoyance for users.